biosimulations/apps/dispatch-service/src/app/app.controller.ts

```diff
@@ -123,22 +134,45 @@ export class AppController {
                     chartJsonPath,
                     JSON.stringify(chartResults)
                   ).then(() => {
-                    // TODO: place all message patterns in a single location
-
-                    this.messageClient.emit('dispatch_result', {
+                    this.messageClient.emit(MQDispatch.SIM_RESULT_FINISH, {
                       uuid: true,
                     });
                   });
                 });
+              })
+              .on('error', (err) => {
+                console.log('Error occured in file writing', err);
               });
           }
         }
-      })
-
-      
+      });
     }
   }
 
+  @MessagePattern(MQDispatch.SIM_DISPATCH_FINISH)
```

Comment:
Here as well, im confused as to why the finish is when the job is started? 
Does finish refer to the process of dispatching?

Reply:
This message means, execute the below function as soon as the process of dispatching the simulation is finished.
Because it seems more intuitive to write it in this way where we actually emit the message.